src/app/utils/encrypt-submission.ts

```diff
@@ -0,0 +1,56 @@
+import { Request } from 'express'
+
+import { IPopulatedForm } from 'src/types'
+
+import { createLoggerWithLabel } from '../../config/logger'
+import { getProcessedResponses } from '../modules/submission/submission.service'
+
+import { checkIsEncryptedEncoding } from './encryption'
+import { createReqMeta } from './request'
+
+const logger = createLoggerWithLabel(module)
+
+type ReqWithForm<T> = T & { form: IPopulatedForm }
+
+export const isValidEncryptSubmission = (req: Request): boolean => {
+  // TODO (#42): Remove typecast once app has migrated away from middlewares.
+  const { form } = req as ReqWithForm<typeof req>
+  try {
+    // Check if the encrypted content is base64
+    checkIsEncryptedEncoding(req.body.encryptedContent)
+  } catch (error) {
+    logger.error({
+      message: 'Invalid encryption',
+      meta: {
+        action: 'validateEncryptSubmission',
+        ...createReqMeta(req),
+        formId: form._id,
+      },
+      error,
+    })
+    return false
+  }
+  return true
+}
+
+export const processResponses = (req: Request): void => {
```

Comment:
Same here. The argument for this function should not be `Request`, but scoped to the variable it is going to act on. In this case, this whole block seems to be the domain of the controller and does not need to be extracted.

Also strive to make utility functions [pure](https://medium.com/javascript-scene/master-the-javascript-interview-what-is-a-pure-function-d1c076bec976). In this function, we are assigning to `req.body.parsedResponses` as a side effect, which can be hard to realize